docs/design/flow-traffic-mirroring/flow-traffic-mirroring.md

```diff
@@ -0,0 +1,171 @@
+                  "vlan": int
+              }
+          }
+      ]
+  }
+  ```
+
+
+* API that needs to be updated: 
+  - Need to add information about the state of the mirror paths in the flow payload.
+  - It is necessary to add information about built mirror paths to API `GET /flows/{flow_id}/path`
+
+## Workflow
+
+Creation and deletion flow mirror points in the FlowHSTopology:
+
+![Create flow traffic mirror point](./create-mirror-point.png "Create flow traffic mirror point")
+![Delete flow traffic mirror point](./delete-mirror-point.png "Delete flow traffic mirror point")
+
+Getting flow mirror points in the NbWorkerTopology:
+
+![Get flow traffic mirror point](./get-mirror-point.png "Get flow traffic mirror point")
+
+## Affected kilda components
+* need to add mirror groups to the Floodlight;
+* need to update RerouteTopology to react to the network events for the mirror paths;
+* add logic to the flow update operation when updating flow endpoints;
+* update switch and flow validation.
+
+## Limitations
+It is allowed to use a transit switch as a mirror point only if flow is pinned.
+
+## Related issues
+There are currently no asymmetric bandwidth ISLs in the system. This feature sets paths in one direction only. 
+This results in asymmetric ISLs appearing in the system. This effect on the system requires more in-depth research.
+
+A possible solution to this issue is to create a dummy path that will consume bandwidth in the opposite direction.
+
+## Switch rules
+Existing actions set for any type of existing OF flows (ingress, transit, egress) 
+will be replaced with a "goto group" action instead of an "output port" action.
+The group will have 2 or more buckets: one will represent output to the flow or ISL port, 
+the rest will represent mirror actions set (i.e. routing to the mirror paths).
+
+Examples of rules for VLAN transit encapsulation:
+* _Group_:
+```
+{
+   "dpid": dpid,
+   "type": "ALL",
+   "group_id": GROUP_ID,
+   "buckets":[
+      {
+         "actions":[
+            {
+               "type": "OUTPUT",
+               "port": flow_or_isl_out_port
+            }
+         ]
+      },
+      {
+         "actions":[
+            {
+               "type": "SET_FIELD", 
+               "field": "vlan_vid",
+               "value": vid
+            },
+            {
+               "type": "OUTPUT",
+               "port": mirror_out_port
+            }
+         ]
+      }
+   ]
+}
+```
+* _Mirror rule_ for egress rule:
+```
+{
+   "dpid": dpid,
+   "cookie": MIRROR_FLOW_COOKIE,
+   "table_id": EGRESS_TABLE_ID,
+   "priority": priority,
+   "match":{
+      "in_port": in_port,
+      "vlan_vid": transit_vid
+   },
+   "actions":[
+      {
+         "type":"GROUP",
+         "group_id":"GROUP_ID"
+      }
+   ]
+}  
+```
+* _Mirror rule_ for ingress rule:
+```
+{
+   "dpid": dpid,
+   "cookie": MIRROR_FLOW_COOKIE,
+   "table_id": INGRESS_TABLE_ID,
+   "priority": priority,
+   "match":{
+      "metadata": metadata,
```

Comment:
Will/should it work in single table mode?

Reply:
These examples have been removed as they are confusing. An explanation that this feature should work in both single table mode and multi table mode has been added.